File: Ordenacao/ordenacaoJAVA/Ordenacao.java
package Ordenacao.ordenacaoJAVA;

public class Ordenacao {

    public void quicksort(int[] conteudoPraOrdenar, int comeco, int fim) {
        if (comeco < fim) {
            int index = organizacao(conteudoPraOrdenar, comeco, fim);
            quicksort(conteudoPraOrdenar, comeco, index - 1);  
            quicksort(conteudoPraOrdenar, index + 1, fim);  
        }
    }

    private int organizacao(int[] ordenando, int comeco, int fim) {
        int pivo = ordenando[fim]; 
        int i = comeco - 1;    

        for (int j = comeco; j < fim; j++) {
            if (ordenando[j] <= pivo) {
                i++;
                int temp = ordenando[i];
                ordenando[i] = ordenando[j];
                ordenando[j] = temp;
            }
        }

        int temp = ordenando[i + 1];
        ordenando[i + 1] = ordenando[fim];
        ordenando[fim] = temp;

        return i + 1;  
    }

    public void printArray(int[] array) {
        for (int i : array) {
            System.out.print(i + " ");
        }
        System.out.println();
    }
}

File: Entregavel2/JavaScript/ordenacao.js
function quicksort(array, comeco, fim) {
    if (comeco < fim) {
        let index = organizacao(array, comeco, fim);
        quicksort(array, comeco, index - 1); 
        quicksort(array, index + 1, fim); 
    }
}

function organizacao(array, comeco, fim) {
    let pivo = array[fim];
    let indexMenores = comeco - 1;

    for (let atual = comeco; atual < fim; atual++) {
        if (array[atual] <= pivo) {
            indexMenores++;
            [array[indexMenores], array[atual]] = [array[atual], array[indexMenores]];
        }
    }

    [array[indexMenores + 1], array[fim]] = [array[fim], array[indexMenores + 1]];

    return indexMenores + 1;
}

function printArray(array) {
    return array.join(', ');
}

function mostrarOrdenacao() {
    let input = document.getElementById("Ordenacao").value;
    let array = input.split(',').map(Number); 
    
    if (array.some(isNaN)) {
        document.getElementById("resultado").innerHTML = "Por favor, insira valores válidos (números separados por vírgula).";
        return;
    }

    document.getElementById("resultado").innerHTML = "Antes da organização: " + printArray(array);

    quicksort(array, 0, array.length - 1);

    document.getElementById("resultado").innerHTML += "<br>Depois da organização: " + printArray(array);
}

function limparResultado() {
    document.getElementById("resultado").innerHTML = '';
    document.getElementById("Ordenacao").value = '';
}

document.getElementById("calcularOrdenacao").addEventListener("click", mostrarOrdenacao);
document.getElementById("limpar").addEventListener("click", limparResultado);